### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.SMul

### PR DESCRIPTION
**Pull request by antoinelab01**
---

mathlib3port generated Algebra.Order.Smul as a name, but shouldn't it be Algebra.Order.SMul?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by antoinelab01** [comment]
The only errors left concern the `positivity_smul` tactic. I didn't touch it as I don't know anything about tactic writing.

**Comment by LukasMias** [comment]
> The only errors left concern the `positivity_smul` tactic. I didn't touch it as I don't know anything about tactic writing.

I think this is fine to comment out for now with an appropriate porting note, like in Algebra.Order.Hom.Basic and Algebra.Order.Pi currently. Should be ready for review now!

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Algebra/Order/SMul.lean [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+instance Int.orderedSmul [LinearOrderedAddCommGroup M] : OrderedSMul ℤ M :=
+  OrderedSMul.mk'' fun n hn => by
+    cases n
+    · simp only [Int.ofNat_eq_coe, Int.coe_nat_pos, coe_nat_zsmul] at hn⊢
+      exact strictMono_smul_left hn
+    · cases (Int.negSucc_not_pos _).1 hn
+#align int.ordered_smul Int.orderedSmul
+
+-- TODO: `LinearOrderedField M → OrderedSMul ℚ M`
+instance LinearOrderedSemiring.toOrderedSmul {R : Type _} [LinearOrderedSemiring R] :
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/SMul.lean:191]:
```suggestion
instance LinearOrderedSemiring.toOrderedSMul {R : Type _} [LinearOrderedSemiring R] :
```

File: Mathlib/Algebra/Order/SMul.lean [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+  rw [← inv_smul_smul c a, ← inv_smul_smul c b]
+  refine' hlt' _ _ _ hab (pos_of_mul_pos_right _ hc.le)
+  simp only [c.mul_inv, zero_lt_one]
+#align ordered_smul.mk' OrderedSMul.mk'
+
+instance [OrderedSMul 𝕜 M] [OrderedSMul 𝕜 N] : OrderedSMul 𝕜 (M × N) :=
+  OrderedSMul.mk' fun _ _ _ h hc =>
+    ⟨smul_le_smul_of_nonneg h.1.1 hc.le, smul_le_smul_of_nonneg h.1.2 hc.le⟩
+
+instance Pi.orderedSmul {M : ι → Type _} [∀ i, OrderedAddCommMonoid (M i)]
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/SMul.lean:221]: Also elsewhere
```suggestion
instance Pi.orderedSMul {M : ι → Type _} [∀ i, OrderedAddCommMonoid (M i)]
```

**Comment by jcommelin** [comment]
bors d+

**Pull request closed without merging**